extensions/azcli/src/azdata.ts

```diff
@@ -262,451 +240,76 @@ export class AzdataTool implements azdataExt.IAzdataApi {
-	await executeSudoCommand('curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/microsoft.asc.gpg > /dev/null');
-	// Add the azdata repository information
-	const release = (await executeCommand('lsb_release', ['-rs'])).stdout.trim();
-	await executeSudoCommand(`add-apt-repository "$(wget -qO- https://packages.microsoft.com/config/ubuntu/${release}/mssql-server-2019.list)"`);
-	// Update repository information and install azdata
-	await executeSudoCommand('apt-get update');
-	await executeSudoCommand('apt-get install -y azdata-cli');
-}
-
-/**
- */
-async function findSpecificAzdata(): Promise<IAzdataTool> {
-	const path = await ((process.platform === 'win32') ? searchForCmd('azdata.cmd') : searchForCmd('azdata'));
-	const versionOutput = await executeAzdataCommand(`"${path}"`, ['--version']);
-	return new AzdataTool(path, parseVersion(versionOutput.stdout));
-}
-
-function getConfig(key: string): AzdataDeployOption | undefined {
-	const config = vscode.workspace.getConfiguration(azdataConfigSection);
-	const value = <AzdataDeployOption>config.get<AzdataDeployOption>(key);
-	Logger.log(loc.azdataUserSettingRead(key, value));
-	return value;
-}
-
-async function setConfig(key: string, value: string): Promise<void> {
-	const config = vscode.workspace.getConfiguration(azdataConfigSection);
-	await config.update(key, value, vscode.ConfigurationTarget.Global);
-	Logger.log(loc.azdataUserSettingUpdated(key, value));
-}
-
-/**
- * Gets the latest azdata version available for a given platform
- */
-export async function discoverLatestAvailableAzdataVersion(): Promise<SemVer> {
-	Logger.log(loc.checkingLatestAzdataVersion);
-	switch (process.platform) {
-		case 'darwin':
-			return await discoverLatestStableAzdataVersionDarwin();
-		// case 'linux':
-		// ideally we would not to discover linux package availability using the apt/apt-get/apt-cache package manager commands.
-		// However, doing discovery that way required apt update to be performed which requires sudo privileges. At least currently this code path
-		// gets invoked on extension start up and prompt user for sudo privileges is annoying at best. So for now basing linux discovery also on a releaseJson file.
-		default:
-			return await getPlatformReleaseVersion();
-	}
-}
-
-/**
- * Parses out the azdata version from the raw azdata version output
- * @param raw The raw version output from azdata --version
+ * Parses out the Azure CLI version from the raw az version output
+ * @param raw The raw version output from az --version
  */
 function parseVersion(raw: string): string {
 	// Currently the version is a multi-line string that contains other version information such
-	// as the Python installation, with the first line being the version of azdata itself.
-	const lines = raw.split(os.EOL);
-	return lines[0].trim();
+	// as the Python installation, with the first line holding the version of az itself.
```

Comment:
Would suggest giving an example output in the comment here so others can see what the format looks like